### DISCUSSION TITLE
How do I use the clearColor parameter in V9?

### DISCUSSION THREAD

**Discussion by tjni** [visgl/deck.gl#8808] [Q&A] [answered]
In my app in deck.gl V8, we currently have the following code:
```
<DeckGL
  ...
  parameters: {{
    clearColor: [0, 0.19, 0.27, 0.6],
    blending: true,
    blendFunc: [GL.SRC_COLOR, GL.CONSTANT_ALPHA],
  }}
/>
```

In deck.gl V9, this code no longer type checks and doesn't appear to do anything even if ignoring type checks. Is this a bug or is there a better way to do this that I can migrate to?

**Comment by Pessimistress** [comment, marked as answer]
`clearColor` is no longer supported. You can set the backgroundColor of your canvas instead.

**Reply by tjni** [reply]
Thank you for the pointer. There is still a difference in the final colors, but it's hard for me to tell why. However, the old colors we had were not picked rigorously, so with sufficient tweaking I'm able to get it close to where I want it to be.

To set the background color on the canvas, however, I did have to use a ref to the canvas. If the DeckGL component were to pass background styles to the canvas instead of the container, this would be easier. Is it a change that is possible?

**Reply by chrisgervang** [reply]
There are use cases around image and video export where a canvas backgroundColor won't be composited into the final output, right? I think losing `clearColor` support means we'd need an alternative solution for defining a "sky" color in these export cases.

**Reply by ibgreen** [reply]
The natural place to specify clear color is in the [view](https://deck.gl/docs/api-reference/core/view#clear). Each view will set up a RenderPass, and since RenderPass initialization is where clearing is done in WebGPU, deck.gl is aligning with that.

**Reply by Pessimistress** [reply]
Created an issue: https://github.com/visgl/deck.gl/issues/8838
